src/plugins/intel_gpu/tests/transformations/reduce_modification_for_false_keepdims_test.cpp

```diff
@@ -0,0 +1,173 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <ngraph/function.hpp>
+#include <ngraph/opsets/opset10.hpp>
+#include <ngraph/pass/manager.hpp>
+#include <plugin/transformations/reduce_modification_for_false_keepdims.hpp>
+#include <string>
+#include <transformations/init_node_info.hpp>
+#include <transformations/utils/utils.hpp>
+#include <tuple>
+
+#include "test_utils.h"
+#include "common_test_utils/ngraph_test_utils.hpp"
+
+using namespace testing;
+using namespace ::tests;
+
+using InputShape = ngraph::PartialShape;
+using KeepDims = bool;
+using ReduceAxes = std::vector<int64_t>;
+using ReduceType = std::shared_ptr<ngraph::Node>;
+using ReshapeShape = std::vector<size_t>;
+using NeedReshape = bool;
+
+class ReduceModificationTests
+    : public ::testing::Test,
+      public testing::WithParamInterface<
+          std::tuple<ReduceType, InputShape, ReduceAxes, KeepDims, NeedReshape, ReshapeShape>> {
+public:
+    std::shared_ptr<ngraph::Function> f, f_ref;
+
+    void SetUp() override {
+        const auto& reduce_type = std::get<0>(GetParam());
+        const auto& input_shape = std::get<1>(GetParam());
+        const auto& axes = std::get<2>(GetParam());
+        const auto& keep_dims = std::get<3>(GetParam());
+        const auto& need_reshape = std::get<4>(GetParam());
+        const auto& reshape_shape = std::get<5>(GetParam());
+
+        f = get_initial_function(input_shape, axes, reduce_type, keep_dims, need_reshape, reshape_shape);
+        f_ref = get_reference_function(input_shape, axes, reduce_type);
+    }
+
+    static std::shared_ptr<ngraph::Function> get_reference_function(const ngraph::PartialShape& input_shape,
+                                                                    const std::vector<int64_t>& axes,
+                                                                    const ReduceType& reduce_type) {
```

Comment:
`reduce_type` is not used

Reply:
It is modified to cldnn::reduce_mode. And it is used  with newly added TCs.

Reply:
relevant  TC : positive :
ReduceDecomposeForFalseKeepdims/ReduceDecomposeTests.CompareFunctions/1
ReduceDecomposeForFalseKeepdims/ReduceDecomposeTests.CompareFunctions/2
ReduceDecomposeForFalseKeepdims/ReduceDecomposeTests.CompareFunctions/3
negative :
ReduceDecomposeForFalseKeepdimsNotCase/ReduceDecomposeTests.CompareFunctions/4
ReduceDecomposeForFalseKeepdimsNotCase/ReduceDecomposeTests.CompareFunctions/5